domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -922,6 +933,52 @@ public void save(AsyncResult<SQLConnection> sqlConnection, String table, String
     }
   }
 
+  /**
+   * Save entity in table and return the updated version (after database trigger for example).
+   *
+   * @param sqlConnection connection (for example with transaction)
+   * @param table where to insert the entity record
+   * @param id  the value for the id field (primary key)
+   * @param entity  the record to insert, either a POJO or a JsonArray, see convertEntity
+   * @param upsert whether to update if the record with that id already exists (INSERT or UPDATE)
+   * @param convertEntity true if entity is a POJO, false if entity is a JsonArray
+   * @param replyHandler  where to report success status and the updated entity
+   */
+  private <T> void saveAndReturnUpdatedEntity(AsyncResult<SQLConnection> sqlConnection, String table, String id, T entity,
+      Handler<AsyncResult<T>> replyHandler) {
+
+    log.debug("save (with connection and id) called on " + table);
+
+    if (sqlConnection.failed()) {
+      replyHandler.handle(Future.failedFuture(sqlConnection.cause()));
+      return;
+    }
+
+    try {
+      long start = System.nanoTime();
+      String sql = INSERT_CLAUSE + schemaName + DOT + table
+          + " (id, jsonb) VALUES (?, ?::JSON) RETURNING jsonb";
+      JsonArray jsonArray = new JsonArray().add(id).add(pojo2json(entity));
+      sqlConnection.result().queryWithParams(sql, jsonArray, query -> {
+          statsTracker(SAVE_STAT_METHOD, table, start);
+          if (query.failed()) {
+            replyHandler.handle(Future.failedFuture(query.cause()));
```

Comment:
Can you avoid the else clause by adding `return;` to the end of this if clause? Using such a guard clause will reduce the indentation of the code in the (former) else clause and will improve readability.